File: tests/fixtures/nestjs/src/auth.controller.ts
import { Controller, Get, Post, Put, Delete, Patch, Param, Body, Query, HttpCode, HttpStatus, Headers } from '@nestjs/common';

@Controller('auth')
export class AuthController {
  @Post('login')
  @HttpCode(HttpStatus.OK)
  login(@Body() loginDto: any): string {
    return 'User logged in';
  }

  @Post('register')
  @HttpCode(HttpStatus.CREATED)
  register(@Body() registerDto: any): string {
    return 'User registered';
  }

  @Post('logout')
  @HttpCode(HttpStatus.OK)
  logout(@Headers('authorization') token: string): string {
    return 'User logged out';
  }

  @Post('refresh')
  refreshToken(@Body() refreshDto: any): string {
    return 'Token refreshed';
  }

  @Post('forgot-password')
  forgotPassword(@Body() forgotPasswordDto: any): string {
    return 'Password reset email sent';
  }

  @Post('reset-password')
  resetPassword(@Body() resetPasswordDto: any): string {
    return 'Password reset successfully';
  }

  @Post('verify-email')
  verifyEmail(@Body() verifyEmailDto: any): string {
    return 'Email verified';
  }

  @Post('resend-verification')
  resendVerification(@Body() resendDto: any): string {
    return 'Verification email resent';
  }

  @Get('profile')
  getProfile(@Headers('authorization') token: string): string {
    return 'User profile';
  }

  @Put('profile')
  updateProfile(
    @Headers('authorization') token: string,
    @Body() updateProfileDto: any
  ): string {
    return 'Profile updated';
  }

  @Patch('password')
  changePassword(
    @Headers('authorization') token: string,
    @Body() changePasswordDto: any
  ): string {
    return 'Password changed';
  }

  @Get('sessions')
  getActiveSessions(@Headers('authorization') token: string): string {
    return 'Active sessions';
  }

  @Delete('sessions')
  terminateAllSessions(@Headers('authorization') token: string): string {
    return 'All sessions terminated';
  }

  @Delete('sessions/:sessionId')
  terminateSession(
    @Headers('authorization') token: string,
    @Param('sessionId') sessionId: string
  ): string {
    return `Session ${sessionId} terminated`;
  }

  @Get('permissions')
  getUserPermissions(@Headers('authorization') token: string): string {
    return 'User permissions';
  }

  @Post('2fa/enable')
  enable2FA(@Headers('authorization') token: string): string {
    return '2FA enabled';
  }

  @Post('2fa/disable')
  disable2FA(
    @Headers('authorization') token: string,
    @Body() disable2FADto: any
  ): string {
    return '2FA disabled';
  }

  @Post('2fa/verify')
  verify2FA(@Body() verify2FADto: any): string {
    return '2FA verified';
  }

  @Get('security-log')
  getSecurityLog(@Headers('authorization') token: string): string {
    return 'Security log';
  }
}

File: tests/fixtures/spring/src/main/java/com/example/AuthController.java
package com.example;

import org.springframework.web.bind.annotation.*;
import org.springframework.http.ResponseEntity;
import org.springframework.http.HttpStatus;

@RestController
@RequestMapping(path = "/auth")
public class AuthController {
    
    @PostMapping("/login")
    public ResponseEntity<String> login(@RequestBody Object loginDto) {
        return ResponseEntity.ok("User logged in");
    }
    
    @PostMapping(value = "/register")
    public ResponseEntity<String> register(@RequestBody Object registerDto) {
        return ResponseEntity.status(HttpStatus.CREATED).body("User registered");
    }
    
    @PostMapping(path = "/logout")
    public ResponseEntity<String> logout(@RequestHeader("Authorization") String token) {
        return ResponseEntity.ok("User logged out");
    }
    
    @RequestMapping(value = "/refresh", method = RequestMethod.POST)
    public ResponseEntity<String> refreshToken(@RequestBody Object refreshDto) {
        return ResponseEntity.ok("Token refreshed");
    }
    
    @PostMapping("/forgot-password")
    public ResponseEntity<String> forgotPassword(@RequestBody Object forgotPasswordDto) {
        return ResponseEntity.ok("Password reset email sent");
    }
    
    @PostMapping(value = "/reset-password")
    public ResponseEntity<String> resetPassword(@RequestBody Object resetPasswordDto) {
        return ResponseEntity.ok("Password reset successfully");
    }
    
    @PostMapping(path = "/verify-email")
    public ResponseEntity<String> verifyEmail(@RequestBody Object verifyEmailDto) {
        return ResponseEntity.ok("Email verified");
    }
    
    @RequestMapping(value = "/resend-verification", method = RequestMethod.POST)
    public ResponseEntity<String> resendVerification(@RequestBody Object resendDto) {
        return ResponseEntity.ok("Verification email resent");
    }
    
    @GetMapping("/profile")
    public ResponseEntity<String> getProfile(@RequestHeader("Authorization") String token) {
        return ResponseEntity.ok("User profile");
    }
    
    @GetMapping(value = "/permissions")
    public ResponseEntity<String> getUserPermissions(@RequestHeader("Authorization") String token) {
        return ResponseEntity.ok("User permissions");
    }
    
    @RequestMapping(value = "/sessions", method = RequestMethod.GET)
    public ResponseEntity<String> getActiveSessions(@RequestHeader("Authorization") String token) {
        return ResponseEntity.ok("Active sessions");
    }
    
    @PutMapping("/profile")
    public ResponseEntity<String> updateProfile(@RequestHeader("Authorization") String token,
                                              @RequestBody Object updateProfileDto) {
        return ResponseEntity.ok("Profile updated");
    }
    
    @PutMapping(value = "/preferences")
    public ResponseEntity<String> updatePreferences(@RequestHeader("Authorization") String token,
                                                   @RequestBody Object preferencesDto) {
        return ResponseEntity.ok("Preferences updated");
    }
    
    @PatchMapping("/password")
    public ResponseEntity<String> changePassword(@RequestHeader("Authorization") String token,
                                                @RequestBody Object changePasswordDto) {
        return ResponseEntity.ok("Password changed");
    }
    
    @PatchMapping(value = "/email")
    public ResponseEntity<String> changeEmail(@RequestHeader("Authorization") String token,
                                            @RequestBody Object changeEmailDto) {
        return ResponseEntity.ok("Email changed");
    }
    
    @RequestMapping(value = "/2fa/enable", method = RequestMethod.PATCH)
    public ResponseEntity<String> enable2FA(@RequestHeader("Authorization") String token) {
        return ResponseEntity.ok("2FA enabled");
    }
    
    @DeleteMapping("/sessions")
    public ResponseEntity<String> terminateAllSessions(@RequestHeader("Authorization") String token) {
        return ResponseEntity.ok("All sessions terminated");
    }
    
    @DeleteMapping(value = "/sessions/{sessionId}")
    public ResponseEntity<String> terminateSession(@RequestHeader("Authorization") String token,
                                                  @PathVariable String sessionId) {
        return ResponseEntity.ok("Session " + sessionId + " terminated");
    }
    
    @RequestMapping(value = "/account", method = RequestMethod.DELETE)
    public ResponseEntity<String> deleteAccount(@RequestHeader("Authorization") String token,
                                               @RequestBody Object deleteAccountDto) {
        return ResponseEntity.ok("Account deleted");
    }
}